cli/dotfiles.go

```diff
@@ -193,6 +193,18 @@ func (r *RootCmd) dotfiles() *clibase.Cmd {
 				}
 
 				_, _ = fmt.Fprintf(inv.Stdout, "Running %s...\n", script)
+
+				// Check if the script is executable and notify on error
+				scriptPath := filepath.Join(dotfilesDir, script)
+				fi, err := os.Stat(scriptPath)
+				if err != nil {
+					return xerrors.Errorf("stat %s: %w", scriptPath, err)
+				}
+
+				if fi.Mode()&0o111 == 0 {
+					return xerrors.Errorf("%[1]q script is not executable. You can solve this by making it executable in your dotfiles repo:\n  chmod +x %[1]q && git add %[1]q && git commit -m \"Make %[1]s executable\" && git push", script)
```

Comment:
```suggestion
					return xerrors.Errorf("%[1]q script is not executable. Please check https://coder.com/docs/v2/latest/dotfiles, script)
```
@mafredri suggested to add docs and then linking to docs.

Reply:
Yeah makes sense, should not repeat it